### PR TITLE
Create add meeting to room form

### PR DESCRIPTION
**Pull request by Leonils**
## Purpose

A form to create a new meeting. It comes with a few new inputs components and refactoring

Screenshot from 2022-07-07 18-04-09

### CODE REVIEW

File: src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  return (
+    <Box
+      border={{ side: 'all', color: 'brand' }}
+      round="8px"
+      pad={{ vertical: 'xsmall', horizontal: 'small' }}
+      margin={margin}
+    >
+      <Box margin={{ left: '11px' }}>
+        <label htmlFor={name}>
+          <Text size="xsmall" weight="bold" color="brand">
```
Comment (kernicPanel) [inline src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx:33]: Maybe the label should be also red if required ?
```suggestion
          <Text size="xsmall" weight="bold" color={required ? 'status-error' : 'brand'}>
```

File: src/frontend/magnify/src/hooks/useFormState.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ interface UseFormState<T> {
  *      As input, each validator receive the new value of the field, and the values of all other fields.
  * @returns UseFormState
  */
-export default function useFormState<T extends Record<string, string>>(
+export default function useFormState<T extends Record<string, string | boolean | number>>(
```
Comment (kernicPanel) [inline src/frontend/magnify/src/hooks/useFormState.ts:49]: Maybe this change should be reflected to the documentation above ?

File: src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx (hunk 1 above)
Comment (Leonils) [reply src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx:33]: Isn't it too violent ? I see the red start on a lot of website, but the whole label... I don't know...

File: src/frontend/magnify/src/hooks/useFormState.ts (hunk 2 above)
Comment (Leonils) [reply src/frontend/magnify/src/hooks/useFormState.ts:49]: Ok. I'll look at it tomorrow. Thank you

File: src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx (hunk 1 above)
Comment (kernicPanel) [reply src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx:33]: Maybe red is too `errorish` but I think the label should have the same color than the appended `*` from lines 37-39
Maybe use `status-warning` or something else for both ?

Comment (Leonils) [reply src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx:33]: I don't think I saw this a lot before. For the sake of the demonstration, here are a dummy form with required and not required fields with each option. I'm not very comfortable with last 2 options, but maybe we can ask the designer ? What do you think ?

Screenshot from 2022-07-08 09-22-53
Screenshot from 2022-07-08 09-23-32
Screenshot from 2022-07-08 09-23-50

Comment (kernicPanel) [reply src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx:33]: This was just a proposal, let's keep the label on his initial color :+1:

Comment (kernicPanel) [reply src/frontend/magnify/src/components/design-system/Fieldset/Fieldset.tsx:33]: Thanks for taking time to do some tries :)

File: src/frontend/magnify/src/hooks/useFormState.ts (hunk 2 above)
Comment (kernicPanel) [reply src/frontend/magnify/src/hooks/useFormState.ts:50]: I'm unresolving this, just for not forgetting.

**Review by kernicPanel — COMMENTED** [review]
Still one conversation to resolve, and we are good to go :+1:

File: src/frontend/magnify/src/hooks/useFormState.ts (hunk 2 above)
Comment (Leonils) [reply src/frontend/magnify/src/hooks/useFormState.ts:50]: I just fixed the wrong commit. This should be fixed now for the PR. The fixed commit is 18e15ca1

**Review by kernicPanel — APPROVED** [review]

**Pull request merged**